src/operator/controllers/svc_reconcilers/svc_reconcilers.go

```diff
@@ -0,0 +1,153 @@
+package svc_reconcilers
+
+import (
+	"context"
+	"fmt"
+	"github.com/amit7itz/goset"
+	otterizev1alpha2 "github.com/otterize/intents-operator/src/operator/api/v1alpha2"
+	"github.com/otterize/intents-operator/src/operator/controllers/intents_reconcilers"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/samber/lo"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type ServiceWatcher struct {
+	client.Client
+	networkPolicyReconciler *intents_reconcilers.NetworkPolicyReconciler
+	injectablerecorder.InjectableRecorder
+}
+
+func NewServiceWatcher(c client.Client, eventRecorder record.EventRecorder, networkPolicyReconciler *intents_reconcilers.NetworkPolicyReconciler) *ServiceWatcher {
+	recorder := injectablerecorder.InjectableRecorder{Recorder: eventRecorder}
+	return &ServiceWatcher{
+		Client:                  c,
+		InjectableRecorder:      recorder,
+		networkPolicyReconciler: networkPolicyReconciler,
+	}
+}
+
+func (r *ServiceWatcher) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// We should reconcile everytime a service changes.
+	// When a target port changes netpols should be updated, etc.
+	service := corev1.Service{}
+	err := r.Get(ctx, req.NamespacedName, &service)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return ctrl.Result{}, err
+	}
+
+	targetServerName := fmt.Sprintf("svc:%s.%s", req.Name, req.Namespace)
+	var intentsList otterizev1alpha2.ClientIntentsList
+	err = r.List(
+		ctx, &intentsList,
+		&client.MatchingFields{otterizev1alpha2.OtterizeTargetServerIndexField: targetServerName})
```

Comment:
What about svc:svcname without namespace?

Reply:
Should List for ClientIntents in same namespace of Service that refer to service name without the namespace suffix.